File: cpp/560.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, int k) {
        int size = nums.size(), ret = 0, sum = 0;
        unordered_map<int, int> m{{0,1}};
        for (int i = 0; i < size; ++i) {
            sum += nums[i];
            ret += m[sum-k];
            ++m[sum];
        }
        return ret;
    }
};

File: go/560.go
subarraySum(nums []int, k int) int {
    ret, m, sum := 0, map[int]int{0: 1}, 0
    for _, num := range nums {
        sum += num
        ret += m[sum-k]
        m[sum]++
    }
    return ret
}
